File: UServer/src/game/player/player.lua
-- @Author: fasthro
-- @Date:   2020-11-26 15:14:29
-- @Last Modified by:   fasthro
-- @Last Modified time: 2020-12-25 17:42:51
local player = class("player")

function player:initialize(...)
    self.uid = 0
    self.login_time = 0

    self:init_session()
end

local logics = {"login", "session"}
for _, v in ipairs(logics) do
	local name = "player.playerlogic." .. v
	local logic = require(name)
	player:include(logic)
end

return player

File: LockstepServer/LockstepServer/Src/Game/Player/Player.cs
/*
 * @Author: fasthro
 * @Date: 2020/12/23 14:03:25
 * @Description:
 */

namespace LockstepServer.Src
{
    public class Player : IPlayer
    {
        #region private

        private long _uid;
        private ISession _session;

        #endregion private

        #region interface

        public long uid => _uid;
        public ISession session => _session;

        #endregion interface

        #region data

        public bool isReady { get; set; }

        #endregion data

        public Player(long uid, ISession session)
        {
            _uid = uid;
            _session = session;
        }
    }
}